src/DGtal/geometry/volumes/BoundedLatticePolytope.h

```diff
@@ -0,0 +1,878 @@
+     * @param itB the start of the range of n+1 points defining the simplex.
+     * @param itE past the end the range of n+1 points defining the simplex.
+     */
+    template <typename PointIterator>
+    BoundedLatticePolytope( PointIterator itB, PointIterator itE );
+
+    /**
+     * Constructs a polytope from a domain and a collection of half-spaces.
+     *
+     * @tparam HalfSpaceIterator any model of forward iterator on HalfSpace.
+     * @param domain a bounded lattice domain.
+     * @param itB the start of the range of half-spaces.
+     * @param itE past the end of the range of half-spaces.
+     *
+     * @param valid_edge_constraints when 'true', tells that there are
+     * half-spaces that represents th constraints on edges (n-2 cells)
+     * lying between two faces (n-1 cells) pointing to different
+     * orthants.
+     */
+    template <typename HalfSpaceIterator>
+    BoundedLatticePolytope( const Domain& domain,
+			    HalfSpaceIterator itB, HalfSpaceIterator itE,
+                            bool valid_edge_constraints = false );
+
+    /**
+     * Initializes a polytope from a domain and a vector of half-spaces.
+     *
+     * @tparam HalfSpaceIterator any model of forward iterator on HalfSpace.
+     * @param domain a bounded lattice domain.
+     * @param itB the start of the range of half-spaces.
+     * @param itE past the end of the range of half-spaces.
+     *
+     * @param valid_edge_constraints when 'true', tells that there are
+     * half-spaces that represents th constraints on edges (n-2 cells)
+     * lying between two faces (n-1 cells) pointing to different
+     * orthants.
+     */
+    template <typename HalfSpaceIterator>
+    void init( const Domain& domain,
+	       HalfSpaceIterator itB, HalfSpaceIterator itE,
+               bool valid_edge_constraints = false );
+
+    
+    /**
+     * Initializes the polytope from a simplex given as a range [itB,itE) of points.
+     *
+     * @param itB the start of the range of n+1 points defining the simplex.
+     * @param itE past the end the range of n+1 points defining the simplex.
+     *
+     * @return 'true' if [itB,itE) was a valid simplex, otherwise
+     * return 'false' and the polytope is empty.
+     *
+     * @note Use DGtal SimpleMatrix::determinant which is not efficient when the
+     * dimension is high. Does not use Eigen to avoid dependency.
+     */
+    template <typename PointIterator>
+    bool init( PointIterator itB, PointIterator itE );
+    
+    /**
+     * Assignment.
+     * @param other the object to copy.
+     * @return a reference on 'this'.
+     */
+    Self & operator= ( const Self & other ) = default;
+
+    /// @}
+
+    // ----------------------- Accessor services ------------------------------
+  public:
+    /// @name Accessor services
+    /// @{
+    
+    /// @return the domain of the current polytope.
+    const Domain& getDomain() const;
```

Comment:
domain() or getDomain() ?

Reply:
domain() to be consistent with other classes (as it is just return on a class member)